cast/src/lib.rs

```diff
@@ -149,11 +152,38 @@ pub async fn wait_for_tx(provider: &JsonRpcClient<HttpTransport>, tx_hash: Field
                 sleep(Duration::from_secs(5));
                 true
             }
-            TransactionStatus::AcceptedOnL2 | TransactionStatus::AcceptedOnL1 => false,
+            TransactionStatus::AcceptedOnL2 | TransactionStatus::AcceptedOnL1 => {
+                return Ok("Transaction accepted")
+            }
             TransactionStatus::Rejected => {
-                println!("{}", style("Transaction has been rejected").red());
-                false
+                return Err(anyhow!(
+                    "Transaction has been rejected. Check if max-fee argument is high enough"
+                ));
             }
         }
     } {}
+
+    Err(anyhow!("Should not reach this line"))
+}
+
+#[must_use]
+pub fn get_rpc_error_message(error: StarknetError) -> &'static str {
+    match error {
+        StarknetError::FailedToReceiveTransaction => "Node failed to receive transaction",
+        StarknetError::ContractNotFound => "There is no contract at the specified address",
+        StarknetError::BlockNotFound => "Block was not found",
+        StarknetError::TransactionHashNotFound => {
+            "Transaction with provided hash was not found (does not exist)"
+        }
+        StarknetError::InvalidTransactionIndex => "There is no transaction with such an index",
+        StarknetError::ClassHashNotFound => "Provided class hash does not exist",
+        StarknetError::ContractError => "An error occurred in the called contract",
+        StarknetError::InvalidContractClass => {
+            "Contract class is invalid. Make sure to pass correct one"
+        }
+        StarknetError::ClassAlreadyDeclared => {
+            "Contract with the same class hash is already declared"
+        }
+        _ => "Unknown RPC error",
```

Comment:
I don't think we should be verbose in some cases (eg "Make sure to pass correct one", we can get rid of such things in my opinion). Error messages should be informative, but concise.

although, this is probably an open question, so we can discuss.